docs/guides/deploy-hyperlane-local-agents.mdx

```diff
@@ -0,0 +1,397 @@
+
+To do this, navigate to the agent-configuration at `$CONFIG_FILES` and replace all instances of "localhost" or "127.0.0.1" in to `host.docker.internal`. For example:
+
+```json
+...
+"localnet1": {
+  ...
+  "rpcUrls": [
+    {
+      // "http": "http://localhost:8545"
+      // "http": "http://127.0.0.1:8545"
+      "http": "http://host.docker.internal:8545"
+    }
+  ],
+  ...
+},
+...
+```
+
+**Mounting directories**
+
+Running with Docker adds an extra layer of complexity because config files need to be accessible from within the Docker container, and validator signatures need to be accessible from outside of the container. This can be solved by mounting directories on your file system into the container.
+
+In the arguments below, we:
+
+1. Set the `$CONFIG_FILES` environment variable to a fixed path within the container.
+1. Mount the agent config file to this fixed path and making it readonly.
+1. Mount the persistent data directory at a fixed path within the container.
+1. Mount the validator signatures directory to a fixed path within the container.
+
+```sh
+...
+-e CONFIG_FILES=/config/agent-config.json \
+--mount type=bind,source=$CONFIG_FILES,target=/config/agent-config.json,readonly \
+--mount type=bind,source="$(pwd)"/hyperlane_db_validator_<your_chain_name>,target=/hyperlane_db \
+--mount type=bind,source="$(pwd)"/$VALIDATOR_SIGNATURES_DIR,target=/tmp/validator-signatures \
+...
+```
+
+Hardcoding these paths deduplucates the configuration, making it easier to pass the right arguments when running the container. See the example below, where that's left is for you to configure the chain name and validator key.
+
+```sh
+...
+./validator \
+--db /hyperlane_db \
+--originChainName <your_chain_name> \
+--checkpointSyncer.type localStorage \
+--checkpointSyncer.path /tmp/validator-signatures \
+--validator.key <your_validator_key>
+...
+```
+
+</TabItem>
+<TabItem value="from-source" label="Building from source">
+
+**Clone and setup**
+
+First, clone the Hyperlane monorepo:
+
+```sh
+git clone git@github.com:hyperlane-xyz/hyperlane-monorepo.git
+```
+
+Then follow the [setup instructions](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/README.md) in the `rust` directory. This should setup `rustup` as well as Rosetta 2 if you are on Apple Silicon.
+
+```sh
+# install rustup
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+
+# (apple silicon only) install rosetta 2
+softwareupdate --install-rosetta --agree-to-license
+```
+
+</TabItem>
+</Tabs>
+
+### Run
+
+<Tabs groupId="docker">
+<TabItem value="docker" label="Using Docker">
+Now that you understand more about configuring validator arguments, pull the latest docker image:
+
+```sh
+docker pull gcr.io/abacus-labs-dev/hyperlane-agent:main
```

Comment:
hmm not exactly sure we should be recommending the latest image since a new one is built automatically each time a PR is merged onto `main`, and it may not have been tested in production. Let's stick with whatever image was in the docs previously